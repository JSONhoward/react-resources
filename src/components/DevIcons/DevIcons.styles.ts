import styled from 'styled-components';

export const StyledDevIcons = styled('div')`
position: absolute;
display: flex;
align-items: flex-end;
justify-content: space-evenly;
height: 100%;
width: 100%;
padding-bottom: 1rem;
`