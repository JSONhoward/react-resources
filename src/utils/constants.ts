import cleverP from '../images/clever-programmer-icon.jpg'
import devEd from '../images/dev-ed-icon.jpg'
import fireship from '../images/fireship-icon.jpg'
import freeCodeCamp from '../images/free-code-camp-icon.jpg'
import netNinja from '../images/net-ninja-icon.jpg'
import webDevSimplified from '../images/web-dev-simplified-icon.jpg'
import rwieruch from '../images/rwieruch-icon.svg'
import sitePoint from '../images/sitepoint.svg'
import smashingMagazine from '../images/smashing-magazine.svg'
import codeCademy from '../images/codecademy.svg'
import udemy from '../images/udemy-icon.svg'
import wesBos from '../images/wes-bos-icon.png'
import medium from '../images/medium-icon.png'
import devTo from '../images/dev-to.svg'

export const webDevChannels: { name: string, id: string, icon: string }[] = [
    {
        id: 'UCFbNIlppjAuEX4znoulh0Cw',
        name: 'Web Dev Simplified',
        icon: webDevSimplified
    },
    {
        id: 'UClb90NQQcskPUGDIXsQEz5Q',
        name: 'Dev Ed',
        icon: devEd
    },
    {
        id: 'UCW5YeuERMmlnqo4oq8vwUpg',
        name: 'Net Ninja',
        icon: netNinja
    },
    {
        id: 'UCsBjURrPoezykLs9EqgamOA',
        name: 'Fireship',
        icon: fireship
    },
    {
        id: 'UC8butISFwT-Wl7EV0hUK0BQ',
        name: 'Free Code Camp',
        icon: freeCodeCamp
    },
    {
        id: 'UCqrILQNl5Ed9Dz6CGMyvMTQ',
        name: 'Clever Programmer',
        icon: cleverP
    }
]

export const mockArray: {id: string, url: string, title: string, description: string, thumb: string}[] = [
    {
        id: 'someId',
        url: 'https://www.youtube.com/watch?v=Faz6jkFTaa0',
        title: 'title goes here',
        description: 'lorem ipsum',
        thumb: `https://via.placeholder.com/300x200`
    },
    {
        id: 'someId',
        url: 'https://www.youtube.com/watch?v=Faz6jkFTaa0',
        title: 'title goes here',
        description: 'lorem ipsum',
        thumb: `https://via.placeholder.com/300x200`
    }, {
        id: 'someId',
        url: 'https://www.youtube.com/watch?v=Faz6jkFTaa0',
        title: 'title goes here',
        description: 'lorem ipsum',
        thumb: `https://via.placeholder.com/300x200`
    }, {
        id: 'someId',
        url: 'https://www.youtube.com/watch?v=Faz6jkFTaa0',
        title: 'title goes here',
        description: 'lorem ipsum',
        thumb: `https://via.placeholder.com/300x200`
    },
    {
        id: 'someId',
        url: 'https://www.youtube.com/watch?v=Faz6jkFTaa0',
        title: 'title goes here',
        description: 'lorem ipsum',
        thumb: `https://via.placeholder.com/300x200`
    }
]

export const onlineCourses: {platform: string, name: string, url: string, icon: string}[] = [
    {
        platform: 'FreeCodeCamp',
        name: 'Learn React in 1 Hour by Building a Movie Search App',
        url: 'https://www.freecodecamp.org/news/learn-react-in-1-hour-by-building-a-movie-search-app/',
        icon: freeCodeCamp
    },
    {
        platform: 'Udemy',
        name: 'React - The Complete Guide (incl Hooks, React Router, Redux)',
        url: 'https://www.udemy.com/course/react-the-complete-guide-incl-redux/',
        icon: udemy
    },
    {
        platform: 'CodeCademy',
        name: 'Learn React.js: Part I/Part II',
        url: 'https://www.codecademy.com/learn/react-101',
        icon: codeCademy
    },
    {
        platform: 'none',
        name: 'Fullstack Advanced React & GraphQL',
        url: 'https://advancedreact.com/',
        icon: wesBos
    }
]

export const onlineBlogs: {name: string, url: string, icon: string}[] = [
    {
        name: 'Medium',
        url: 'https://medium.com/',
        icon: medium
    },
    {
        name: 'Dev.to',
        url: 'https://dev.to/',
        icon: devTo
    },
    {
        name: 'Sitepoint',
        url: 'https://www.sitepoint.com/',
        icon: sitePoint
    },
    {
        name: 'Smashing Magazine',
        url: 'https://www.smashingmagazine.com/',
        icon: smashingMagazine
    },
    {
        name: 'RobinWieruch',
        url: 'https://www.robinwieruch.de/blog',
        icon: rwieruch
    }
]