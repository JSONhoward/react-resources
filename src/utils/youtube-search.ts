const YOUTUBE_SEARCH_URL = (channel: string, limit: string): string => `https://www.googleapis.com/youtube/v3/search?part=snippet&channelId=${channel}${limit}&q=react&type=video&key=${process.env.REACT_APP_YOUTUBE_API}`

export const getVideoList = async (channel: string, limit = 50): Promise<Video[]> => {
    const response = await fetch(YOUTUBE_SEARCH_URL(channel, `&maxResults=${limit.toString()}`))
    const data = await response.json()

    const videos: Video[] = await data['items'].map((result: { id: { videoId: any }; snippet: { title: any; description: any; thumbnails: { high: { url: any } } } }) => {
        return {
            id: result.id.videoId,
            url: `http://www.youtube.com/watch?v=${result.id.videoId}`,
            title: result.snippet.title,
            description: result.snippet.description,
            thumb: result.snippet.thumbnails.high.url
        }
    })

    return videos
}