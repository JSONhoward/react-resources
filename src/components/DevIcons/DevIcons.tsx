import React from 'react'

import { StyledDevIcons } from './DevIcons.styles'
import Icon from './Icon'
import { webDevChannels } from '../../utils/constants';

const DevIcons = () => {
    const icons = webDevChannels.map((el, i) => {
        return (
            <Icon key={i} icon={el.icon} title={el.name} />
        )
    })

    return (
        <StyledDevIcons>
            {icons}
        </StyledDevIcons>
    )
}

export default DevIcons
