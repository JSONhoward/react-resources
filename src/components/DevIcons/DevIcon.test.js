import React from 'react'
import {create} from 'react-test-renderer'
import {screen, render} from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import DevIcons from './DevIcons'
import { webDevChannels } from '../../utils/constants'

describe('DevIcon Tests', () => {
    test('Snapshot Test', () => {
        const tree = create(<DevIcons />).toJSON()

        expect(tree).toMatchSnapshot()
    })

    test('renders each dev icon', () => {
        render(<DevIcons />)

        webDevChannels.forEach((dev, i) => {
            expect(screen.getByTestId(`devIcon${dev.name}`))
        })
    })
})