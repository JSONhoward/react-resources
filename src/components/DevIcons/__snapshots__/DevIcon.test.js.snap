// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DevIcon Tests Snapshot Test 1`] = `
<div
  className="sc-bdnylx dNXLBy"
>
  <div
    className="sc-gtssRu kliTqT"
  >
    <img
      alt="Web Dev Simplified"
      data-testid="devIconWeb Dev Simplified"
      src="web-dev-simplified-icon.jpg"
    />
  </div>
  <div
    className="sc-gtssRu kliTqT"
  >
    <img
      alt="Dev Ed"
      data-testid="devIconDev Ed"
      src="dev-ed-icon.jpg"
    />
  </div>
  <div
    className="sc-gtssRu kliTqT"
  >
    <img
      alt="Net Ninja"
      data-testid="devIconNet Ninja"
      src="net-ninja-icon.jpg"
    />
  </div>
  <div
    className="sc-gtssRu kliTqT"
  >
    <img
      alt="Fireship"
      data-testid="devIconFireship"
      src="fireship-icon.jpg"
    />
  </div>
  <div
    className="sc-gtssRu kliTqT"
  >
    <img
      alt="Free Code Camp"
      data-testid="devIconFree Code Camp"
      src="free-code-camp-icon.jpg"
    />
  </div>
  <div
    className="sc-gtssRu kliTqT"
  >
    <img
      alt="Clever Programmer"
      data-testid="devIconClever Programmer"
      src="clever-programmer-icon.jpg"
    />
  </div>
</div>
`;
